### DISCUSSION TITLE
Birt Metaspace usage

### DISCUSSION THREAD

**Discussion by marcelocastrillo** [eclipse-birt/birt#1442] [Q&A]
I am using Birt Report 4.4.2 to generate transactional reports in PDF format. Currently, the service receives approximately 50 requests per minute. The problem is that over time, the response times and PDF generation speed start to decrease, as can be seen in the graph:
![image](https://github.com/eclipse-birt/birt/assets/16979318/503e7310-8d9a-4bd0-a78b-1288b101a4c7)

Initially, I thought it might be a problem with heap memory usage, but I don't see the same pattern there:

![image](https://github.com/eclipse-birt/birt/assets/16979318/4e0d58f8-9247-4733-b39a-2d32905a7243)

I do see a similar pattern in non-heap memory usage.

![image](https://github.com/eclipse-birt/birt/assets/16979318/118f6f14-615a-42d4-9de2-60e978f136d0)

And after the GC collection it is normal again

![image](https://github.com/eclipse-birt/birt/assets/16979318/966e8687-1b64-4b55-9155-eb110913d3c7)

So, my question is why do metaspace grows so fast and if there is a way to avoid this and mantain a stable response time.

**Comment by speckyspooky** [comment]
One of the main unkown on your example and documentation is that your process in which way you have BIRT integrated isn't present.
So I think you will have a kind of Java-program integration.

Therefor your results could be have multiple effects based on your Java-integration and may be BIRT.
On my integration framing we have lot of parallel report requests and yes we have a familiar situation but no prblem with it
because the reports and the used resources are very different on my side.

The MetaSpace handling is a new topic for me, so I read the following side: https://stuefe.de/posts/metaspace/what-is-metaspace/
And it seems to me that it could have the effect like your documentation.

If you have some problems whith it you can try do define limitation with the following configuartion options:

-XX:MaxMetaspaceSize
-XX:CompressedClassSpaceSize

Perhaps it will help a little bit on your research.
